src/System.IO.Pipelines/WritableBufferWriter.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.IO.Pipelines
+{
+    public struct WritableBufferWriter
+    {
+        private WritableBuffer _writableBuffer;
+        private Span<byte> _span;
+
+        public WritableBufferWriter(WritableBuffer writableBuffer)
+        {
+            _writableBuffer = writableBuffer;
+            _span = writableBuffer.Buffer.Span;
+        }
+
+        public Span<byte> Span => _span;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Advance(int count)
```

Comment:
One major difference between this and the `ReadableBufferReader` is that advice doesn't support walking multiple blocks.